snuba/subscriptions/data.py

```diff
@@ -175,68 +188,98 @@ def to_dict(self) -> Mapping[str, Any]:
+                Literal(None, (timestamp - self.time_window)),
+            ),
+            binary_condition(
+                ConditionFunctions.LT,
+                Column(None, None, required_timestamp_column),
+                Literal(None, timestamp),
+            ),
+        ]
 
-            conditions_to_add: List[Expression] = [
-                binary_condition(
-                    ConditionFunctions.EQ,
-                    Column(None, None, "project_id"),
-                    Literal(None, self.project_id),
-                ),
-                binary_condition(
-                    ConditionFunctions.GTE,
-                    Column(None, None, required_timestamp_column),
-                    Literal(None, (timestamp - self.time_window)),
-                ),
+        if offset is not None:
+            conditions_to_add.append(
                 binary_condition(
-                    ConditionFunctions.LT,
-                    Column(None, None, required_timestamp_column),
-                    Literal(None, timestamp),
-                ),
-            ]
-
-            if offset is not None:
-                conditions_to_add.append(
-                    binary_condition(
-                        ConditionFunctions.LTE,
-                        FunctionCall(
-                            None,
-                            "ifnull",
-                            (Column(None, None, "offset"), Literal(None, 0)),
-                        ),
-                        Literal(None, offset),
-                    )
+                    ConditionFunctions.LTE,
+                    FunctionCall(
+                        None,
+                        "ifnull",
+                        (Column(None, None, "offset"), Literal(None, 0)),
+                    ),
+                    Literal(None, offset),
                 )
+            )
 
-            new_condition = combine_and_conditions(conditions_to_add)
-            condition = query.get_condition()
-            if condition:
-                new_condition = binary_condition(
-                    BooleanFunctions.AND, condition, new_condition
-                )
+        new_condition = combine_and_conditions(conditions_to_add)
+        condition = query.get_condition()
+        if condition:
+            new_condition = binary_condition(
+                BooleanFunctions.AND, condition, new_condition
+            )
+
+        query.set_ast_condition(new_condition)
+
+    def validate_subscription(
+        self, query: Union[CompositeQuery[Entity], Query]
+    ) -> None:
+        # TODO: Support composite queries with multiple entities.
+        from_clause = query.get_from_clause()
+        if not isinstance(from_clause, Entity):
+            raise InvalidSubscriptionError("Only simple queries are supported")
+        entity = get_entity(from_clause.key)
+
+        SubscriptionAllowedClausesValidator().validate(query)
+        OneProjectValidator().validate(query)
```

Comment:
Why do we need this validator? 
The add_conditions method adds a `project_id = a_value` condition. This means we are always structurally certain there will be a project_id condition that will not be too wide. 
If somebody adds a condition on multiple projects, the query will simply return nothing as it does today.

Reply:
OK sounds good. I had a note from our discussions saying we should validate this but I'll remove it.

Reply:
Yeah I remember that discussion. Sorry I forgot we were actually adding the condition every time we would actually run this validator.

Reply:
It's removed.